sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3566,11 +3564,26 @@ object GpuOverrides extends Logging {
       // This needs to match what murmur3 supports.
       PartChecks(RepeatingParamCheck("hash_key",
         (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 +
-            TypeSig.STRUCT).nested(), TypeSig.all)),
+            TypeSig.STRUCT + TypeSig.ARRAY).nested(),
```

Comment:
Why was murmur3 not updated to indicate that it too supports this?  Also why were the hash partitioning tests not updated?

Reply:
HashPartitioning is being tested by hash_aggregate_test.py is there a place for dedicated tests for HashPartitioning?

I had missed the `murmur3` change, thanks for pointing it out. Where is this being tested? The only place I could find is a unit test `UnaryOperatorsSuite.scala`

Reply:
> is there a place for dedicated tests for HashPartitioning?

See repart_test, there's a test_hash_repartition_exact that does this.  `repartition` can take a column argument in addition to the number of partitions, and if specified, it will use that column to hash into the partitions.

Reply:
Thanks @jlowe